servlet/src/main/java/io/undertow/servlet/core/BlockingWriterSenderImpl.java

```diff
@@ -221,11 +231,15 @@ private void invokeOnComplete(final IoCallback callback) {
             if (writer.checkError()) {
                 queuedCallback.onException(exchange, this, new IOException());
             } else {
-                inCall = true;
+                sendThread = null;
+                inCall = Thread.currentThread();
                 try {
-                    queuedCallback.onComplete(exchange, this);
+                    callback.onComplete(exchange, this);
```

Comment:
@stuartwdouglas  The same here, aren't we supposed to process queuedCallback in this block?